escrow/token_service.proto

```diff
@@ -0,0 +1,61 @@
+syntax = "proto3";
+
+package escrow;
+
+option java_package = "io.singularitynet.daemon.escrow";
+//It is expected that the user would call the GetChannelState to Determine the Current state of the Channel
+//Based on the usage forecast, the user/client will have to sign for an amount L + U , where L is the last amount Signed
+//and U is the amount based on expected usage.
+//Please be aware that the Signing up an amount upfront ( Pre Paid)  does come with a risk and hence the
+//user must exercise caution on the amount signed specially with new service providers.
+//If there is no need of making concurrent calls then you may consider pay per mode.
+//Using a Token, the Client can now make concurrent calls, which was not supported previously with the pay per mode.
+//However the pay per mode is a lot secure than the pre-paid mode.
+service TokenService {
+    // GetToken method checks the Signature sent and returns a Token
+    // 1) The Signature is valid and has to be signed in the below format
+    //"__MPE_claim_message"+MpeContractAddress+ChannelID+ChannelNonce+SignedAmount
+    //Signature is to let the Service Provider make a claim
+    // 2) Signed amount >= Last amount Signed.
+    //planned amount 10 , usage 5 , this signed 10 cogs
+    //  if Signed amount == Last Signed amount , then check if planned_amount < used_amount
+    //  if Signed amount > Last Signed amount , then update the planned amount = planned amount + Signed Amount
```

Comment:
Not sure what is a `plannned amount` here. Am I correct that in situation when user already spent 5 cogs from channel and he wants to sign upfront he signs 13 cogs for example and this means that `planned_amount` is 13 and `used_amount` is 5 ?

Reply:
thanls @vsbogd , just to be sure
amount deposited in channel = 100 cogs 
amount used = 5 Cogs => the user would have given a sig5 ( signature for 5 cogs) 
so the planned amount so far is 5 cogs and the used amount is 5 cogs so far 

Now the user wishes to Sign for 13 Cogs , the planned amount  = Max Amount Signed 
the planned amount now becomes 13 Cogs as you have mentioned and the used amont is still 5 cogs ( thus the user can now make calls worth 13-5 = 8 Cogs) .

The claims in the current block chain supports planned amount and used/actual amount, so this ties nicely on what the latest MPE contract supports.

the planned amount and used amount against a given channel Id will be maintained in a seprate state in etcd , this way user can contiue to use pay per use / support for concurrent calls simultaneously , the new daemon should smoothly run even on old clients

Reply:
Ok, then I believe the following:
> if Signed amount > Last Signed amount , then update the planned amount = planned amount + Signed Amount

should be replaced by:
if Signed amount > Last Signed amount , then update the planned amount = Signed Amount

Reply:
thanks @vsbogd , corrected the comment